hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/KafkaOffsetGen.java

```diff
@@ -282,6 +301,36 @@ private Long delayOffsetCalculation(Option<String> lastCheckpointStr, Set<TopicP
     return delayCount;
   }
 
+  /**
+   * Get the checkpoint by timestamp.
+   * @param consumer
```

Comment:
can you please add some documentation on whats happening here. format etc. an example would be great.

Reply:
OK, I will add it immediately